### PR TITLE
Add more test for broker jersey bounded thread pool

### PR DESCRIPTION
**Pull request by tibrewalpratik17**
label
- `testing`

This is a small patch to add more unit tests for broker managed async executor provider class.

### CODE REVIEW

File: pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java [hunk 1]
```diff
@@ -92,6 +94,23 @@ public void testGet()
 
     BlockingQueue<Runnable> blockingQueue = threadPoolExecutor.getQueue();
     assertNotNull(blockingQueue);
+    assertTrue(blockingQueue instanceof LinkedBlockingQueue);
+    assertEquals(0, blockingQueue.size());
```
Comment (Jackie-Jiang) [inline pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java:98]: Not introduced in this PR, but can you reverse the arguments? The first argument should be the actual value, and second one is the expected value. I'd suggest also fixing the misuse in other places

Comment (tibrewalpratik17) [reply pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java:98]: @Jackie-Jiang 
I think Junit follows the convention of expected first and actual second 
`void org.junit.Assert.assertEquals(Object expected, Object actual)` 
Do you think we should follow something similar?

Comment (Jackie-Jiang) [reply pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java:98]: We always use `org.testng.Assert` where first value is actual

Comment (tibrewalpratik17) [reply pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java]: Updated in this file accordingly.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**